### PR TITLE
noresm2_5_013_cam6_3_158: Update CAM code to cam6_3_158

### LINKED ISSUE #144: Addition of new sum diagnostics for aerosols
### What is the feature/what would you like to discuss?

Changes to be able to have sum diagnostics for comparison of oslo-aero with tropmam4

### Is there anyone in particular you want to be part of this conversation?

_No response_

### Will this change (regression test) answers?

Yes

### Will you be implementing this enhancement yourself?

Yes

### PR DESCRIPTION
**Pull request by mvertens**
Update noresm_develop CAM code to cam6_3_158

Summary: Update noresm_develop CAM code to cam6_3_158 and point to updated oslo-aero changes as well

Contributors: @mvertens @gold2718 

Reviewers: @gold2718 

Note this PR also requires an associated PR https://github.com/NorESMhub/OSLO_AERO/pull/31. Once this is accept the External_CAM.cfg will be updated here.

Updated aux_cam_noresm test was run and new baselines are at /cluster/shared/noresm/noresm_baselines/cam_develop/noresm2_5_alpha04

Analysis here: https://ns2345k.web.sigma2.no/datalake/olivie/noresm2.5/atm.noresm2.5_atm.v1.0.pdf

### CODE REVIEW

File: Externals_CAM.cfg [hunk 1]
```diff
@@ -51,32 +51,47 @@ required = True
...
 repo_url = https://github.com/NCAR/atmospheric_physics
 local_path = src/atmos_phys
 required = True
 
 [camnor_phys]
 protocol = git
-tag = 2aad4ca
-repo_url = https://github.com/NorESMhub/CAM-Nor-physics
+branch = gold2718_CAM-Nor-physics/noresm
+repo_url = https://github.com/mvertens/CAM-Nor-physics
```
Comment (gold2718) [inline Externals_CAM.cfg]: This needs to be updated before merging.

File: Externals_CAM.cfg [hunk 2]
```diff
@@ -51,32 +51,47 @@ required = True
...
 local_path = src/physics/camnor_phys
 required = True
 
 [oslo_aero]
 protocol = git
-tag = noresm_oslo_aero_v1
-repo_url = https://github.com/NorESMhub/OSLO_AERO
+#branch = feature/UpdateNoresmBranchRebase
+tag = 317e4ff
+repo_url = https://github.com/mvertens/OSLO_AERO
```
Comment (gold2718) [inline Externals_CAM.cfg]: This needs to be updated before merging.

File: bld/build-namelist [hunk 3]
```diff
@@ -570,8 +570,8 @@ my $prescribe_aerosols = $TRUE;
 if ($simple_phys) {$prescribe_aerosols = $FALSE;}
 
 if( $chem =~ /_oslo/){
-$prescribe_aerosols = $FALSE;
-print " ==> Using Oslo aerosols: PRESCRIBED AERO = FALSE (not yet implemented) \n"
+    $prescribe_aerosols = $FALSE;
+    print " ==> Using Oslo aerosols: PRESCRIBED AERO = FALSE (not yet implemented) \n"
```
Comment (gold2718) [inline bld/build-namelist:574]: Is there a plan for this?

File: bld/namelist_files/namelist_definition.xml [hunk 4]
```diff
@@ -4959,13 +5126,14 @@ Default: 'off'
 </entry>
 
 <!--==============================================-->
+<!-- TODO: (mvertens) is this still needed -->
```
Comment (gold2718) [inline bld/namelist_files/namelist_definition.xml:5129]: I'm not sure. I do not see any use in cam_development. However, taking it out now would be a lot of work.

File: cime_config/config_pes.xml
Comment (gold2718) [inline cime_config/config_pes.xml:1]: I think we still want updates from NCAR for this file. Please revert to that version.

File: Externals_CAM.cfg [hunk 5]
```diff
@@ -51,32 +51,47 @@ required = True
...
 local_path = src/physics/camnor_phys
 required = True
 
 [oslo_aero]
 protocol = git
-tag = noresm_oslo_aero_v1
-repo_url = https://github.com/NorESMhub/OSLO_AERO
+#branch = feature/UpdateNoresmBranchRebase
+tag = 7abf42a
+repo_url = https://github.com/mvertens/OSLO_AERO
```
Comment (gold2718) [inline Externals_CAM.cfg]: This needs to be updated before merging.

**Review by gold2718 — CHANGES_REQUESTED** [review]
This looks pretty good, just needs a couple of tweaks.

File: bld/build-namelist (hunk 3 above)
Comment (mvertens) [reply bld/build-namelist:574]: @gold2718 - I have no idea. Probably should be an issue?

File: bld/namelist_files/namelist_definition.xml (hunk 4 above)
Comment (mvertens) [reply bld/namelist_files/namelist_definition.xml:5129]: I agree - again I'd like to move this question down the road.

File: cime_config/config_pes.xml
Comment (mvertens) [reply cime_config/config_pes.xml:1]: what is the version I should revert to?

File: bld/build-namelist (hunk 3 above)
Comment (gold2718) [reply bld/build-namelist:574]: See #167

**Review by gold2718 — APPROVED** [review]
Looks good now

File: Externals_CAM.cfg (hunk 5 above)
Comment (mvertens) [reply Externals_CAM.cfg]: Should this point to a OSLO_AERO and CAM-NOR tags or hashes?

**Comment by mvertens** [comment]
@gold2718 - testing shows bfb. If we are okay with oslo_aero and camnor_phys pointing to hashes I think this PR is ready to be merged and tagged.

**Pull request merged**